hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/MorphingPortableReader.java

```diff
@@ -198,10 +198,22 @@ public short readShort(String fieldName) throws IOException {
     }
 
     @Override
+    public boolean[] readBooleanArray(String fieldName) throws IOException {
+        FieldDefinition fd = cd.getField(fieldName);
+        if (fd == null) {
+            return null;
+        }
+        if (fd.getType() != FieldType.BOOLEAN_ARRAY) {
+            throw new IncompatibleClassChangeError();
```

Comment:
I have been checking the documentation of the PortableReader and it is unclear that a null or empty array is returned in certain cases. Can you add this to the documentation? Can be done by writing it once and copy pasting it to each method.